core/templates/services/debouncer.service.ts

```diff
@@ -27,29 +27,25 @@ export class DebouncerService {
   // be invoked. The function only gets executed after it stops being called
   // for `wait` milliseconds.
   debounce(func: () => void, millisecsToWait: number): () => void {
-    let timeout;
-    let context = this;
-    let args = arguments;
-    let timestamp;
-    let result;
+    let timeout: ReturnType<typeof setTimeout> | null;
+    let context: this | null = this;
+    let timestamp: number;
+    let result: void;
```

Comment:
Why is this a void?

Reply:
As I said that this ``` debounce ``` function is used at only one place in codebase in ``` top-navigation-bar.directive.html ``` and there they pass truncateNavbar function in ``` DebouncerService.debounce(truncateNavbar,500) ``` where truncateNavbar is func and it does not return anything when we call this (``` result =func.apply(context) ```) and also does not take any arguments so I removed params above.

Reply:
But this is s general service file write and this function treats the result in a different way, right?

@nishantwrp, Can you please help us here? (I think the result can be anything and it depends on what the function parameter returns on calling. Do you have any suggestions on what should be the type here?)

  @nishantwrp PTAL

Reply:
But as mentioned in line 29 in ```debouncer.service.ts``` the type of "func" should always be the function that returns void ``` func: ( ) => void ```, so I guess it will always return void so we can use a type of result as void.

Reply:
@DubeySandeep I agree with @atpug22, the func argument is a function that returns `void`. So, I guess it's fine to assign `void` type to `result`